client/src/entrypoints/core.js

```diff
@@ -51,296 +51,302 @@ window.initTagField = initTagField;
-        var displayConfirmation = (
-            !formSubmitted && (alwaysDirty || isDirty)
-        );
-
-        if (displayConfirmation) {
-            event.returnValue = confirmationMessage;
-            return confirmationMessage;
-        }
-    });
+  const $form = $(formSelector);
+  const confirmationMessage = options.confirmationMessage || ' ';
+  const alwaysDirty = options.alwaysDirty || false;
+  let initialData = null;
+  let formSubmitted = false;
+
+  $form.on('submit', () => {
+    formSubmitted = true;
+  });
+
+  // Delay snapshotting the form’s data to avoid race conditions with form widgets that might process the values.
+  // User interaction with the form within that delay also won’t trigger the confirmation message.
+  setTimeout(() => {
+    initialData = $form.serialize();
+  }, 1000 * 10);
+
+  // eslint-disable-next-line consistent-return
+  window.addEventListener('beforeunload', (event) => {
+    const isDirty = initialData && $form.serialize() !== initialData;
+    const displayConfirmation = (
+      !formSubmitted && (alwaysDirty || isDirty)
+    );
+
+    if (displayConfirmation) {
+      // eslint-disable-next-line no-param-reassign
+      event.returnValue = confirmationMessage;
+      return confirmationMessage;
+    }
+  });
 }
 window.enableDirtyFormCheck = enableDirtyFormCheck;
 
-$(function() {
-    // Add class to the body from which transitions may be hung so they don't appear to transition as the page loads
-    $('body').addClass('ready');
-
-    // Enable toggle to open/close nav
-    $(document).on('click', '#nav-toggle', function() {
-        $('body').toggleClass('nav-open');
-        if (!$('body').hasClass('nav-open')) {
-            $('body').addClass('nav-closed');
-        } else {
-            $('body').removeClass('nav-closed');
-        }
-    });
-
-    // Enable toggle to open/close user settings
-    $(document).on('click', '#account-settings', function() {
-        $('.nav-main').toggleClass('nav-main--open-footer');
-        $(this).find('em').toggleClass('icon-arrow-down-after icon-arrow-up-after');
-    });
-
-    // Resize nav to fit height of window. This is an unimportant bell/whistle to make it look nice
-    var fitNav = function() {
-        $('.nav-wrapper').css('min-height', $(window).height());
-    };
-
-    fitNav();
+$(() => {
+  // Add class to the body from which transitions may be hung so they don't appear to transition as the page loads
+  $('body').addClass('ready');
+
+  // Enable toggle to open/close nav
+  $(document).on('click', '#nav-toggle', () => {
+    $('body').toggleClass('nav-open');
+    if (!$('body').hasClass('nav-open')) {
+      $('body').addClass('nav-closed');
+    } else {
+      $('body').removeClass('nav-closed');
+    }
+  });
 
-    $(window).on('resize', function() {
-        fitNav();
-    });
+  // Enable toggle to open/close user settings
+  $(document).on('click', '#account-settings', () => {
```

Comment:
We need to watch out for inline functions that refer to `this`, since the semantics are different for arrow functions - in this case, the arrow icon fails to switch direction as the panel is opened/closed.

Reply:
All fixed now, I redid the linting from scratch but left the arrow function change entirely to ``eslint --fix`` this time and it appears to err on the safe side